telegram/ext/jobqueue.py

```diff
@@ -195,6 +210,46 @@ def run_repeating(self, callback, interval, first=None, context=None, name=None)
         self._put(job, time_spec=first)
         return job
 
+    def run_monthly(self, callback, when, day, context=None, name=None):
+        """Creates a new ``Job`` that runs on a monthly basis and adds it to the queue.
+
+        Args:
+            callback (:obj:`callable`): The callback function that should be executed by the new
+                job. It should take ``bot, job`` as parameters, where ``job`` is the
+                :class:`telegram.ext.Job` instance. It can be used to access its ``Job.context``
+                or change it to a repeating job.
+            when (:obj:`datetime.time`): Time of day at which the job should run. If the timezone
+                (``when.tzinfo``) is ``None``, UTC will be assumed.
+            day (:obj:`int`): Defines the day of the month whereby the job would run. It should
+                be within the range of 1 and 31, inclusive.
+            context (:obj:`object`, optional): Additional data needed for the callback function.
+                Can be accessed through ``job.context`` in the callback. Defaults to ``None``.
+            name (:obj:`str`, optional): The name of the new job. Defaults to
+                ``callback.__name__``.
+
+        Returns:
+            :class:`telegram.ext.Job`: The new ``Job`` instance that has been added to the job
+            queue.
+
+        Notes:
+             This method runs the function every monthly at the given date. This means that
+             if there is no such date within that month, the job would not run.
+
+        """
+
+        if 1 <= day <= 31:
+            dt = datetime.datetime.now(tz=when.tzinfo or _UTC)
+            while dt.day != day:
```

Comment:
While this certainly works, it seems a bit hacky to my. Suggestion: Use `calendar.monthrange()` to determine, if the next month has a day `day`. To get the correct input, you can use `when.moth + 1` or something like `when + timedelta(days=calendar.monthrange(dt.year, dt.month) - dt.day + 1)`, which should automatically increment the year, if necessary.